src/metrics.py

```diff
@@ -0,0 +1,148 @@
+    assert isinstance(references,list), "References must be a list as it is looking at multiple captions"
+    assert isinstance(candidates,list), "Candidates must be a list as it is looking at multiple captions"
+
+    meteor_scores = []
+    for i in range(len(references)):
+        meteor_scores.append(calculate_meteor_score_sentence(references[i],candidates[i]))
+    return sum(meteor_scores)/len(meteor_scores)*100
+
+'''
+Calculates the Rouge-L score for any number of sentences
+https://clementbm.github.io/theory/2021/12/23/rouge-bleu-scores.html#:~:text=In%20its%20simplest%20form%20ROUGE%20score%20is%20the,the%20denominator%20ROUGE%20is%20a%20recall%20oriented%20metric.
+'''
+def calculate_rouge_score(reference,candidate):
+    #Example: [ref1a_sentence, ref1b_sentence] or [[ref1a_sentence, ref1b_sentence], [ref2a_sentence, ref2b_sentence]]
+    #["The cat in the hat", "There is a cat on the mat"] or [["The cat in the hat", "There is a cat on the mat"], ["The cat is not on the mat", "There is a dog on the wall"]]
+    #Example: cand sentence or [cand1_sentence, cand2_sentence]
+    #"The cat sits on the mat" or ["The cat sits on the mat", "The cat is on the mat"]
+
+    rouge=evaluate.load('rouge')
+    assert len(reference) == len(candidate), "The lengths of references and candidates must be the same"
+    if not isinstance(reference[0],list):
+        reference=[[sentence]for sentence in reference]
+    if not isinstance(candidate,list):
+        candidate=[candidate]
+
+    return rouge.compute(predictions=candidate, references=reference)['rougeL']*100
+
+'''
+Using AAC-metrics package to output cider-d score
+'''
+def calculate_cider_d_score(references,candidates):
+    try:
+        # Using shell=True to run the command through the shell
+        result = subprocess.run("aac-metrics-download", shell=True, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+        print("Output:", result.stdout)
+    except subprocess.CalledProcessError as e:
+        print("Error occurred:", e.stderr)
+    if not isinstance(references[0],list):
+        references=[[sentence]for sentence in references]
+    assert not isinstance(candidates,str), "Must be more than 2 candidates to calculate cider-d"
+    assert len(candidates)>1, "Must be more than 2 candidates to calculate cider-d"
+    candidates = preprocess_mono_sents(candidates)
+    mult_references = preprocess_mult_sents(references)
+    corpus_scores, sents_scores = cider_d(candidates, mult_references)
+    return corpus_scores
+
+def test_metrics():
+    candidate= ["a man is speaking", "rain falls"]
+    reference = [["a man speaks.", "someone speaks.", "a man is speaking while a bird is chirping in the background"], ["rain is falling hard on a surface"]]
+    #reference = ['this is small test']
+    #candidate = ['this is a test']
+    print(calculate_cider_d_score(reference, candidate))
+
+
+#test_metrics()
```

Comment:
NP: Create a tests/ folder outside of src (in root project directory) and move this function inside of a test_metrics.py file.